guard/tests/test_command.rs

```diff
@@ -6,31 +6,158 @@ pub(crate) mod utils;
 #[cfg(test)]
 mod test_command_tests {
     use std::io::stdout;
-    use cfn_guard::commands::{RULES, TEST, TEST_DATA};
-    use cfn_guard::Error;
+
     use rstest::rstest;
+
+    use cfn_guard::commands::{
+        ALPHABETICAL, DIRECTORY, LAST_MODIFIED, PREVIOUS_ENGINE, RULES, RULES_AND_TEST_FILE,
+        RULES_FILE, TEST, TEST_DATA, VERBOSE,
+    };
     use cfn_guard::utils::writer::WriteBuffer::Stdout;
     use cfn_guard::utils::writer::Writer;
-    use crate::utils::{get_full_path_for_resource_file, cfn_guard_test_command};
+    use cfn_guard::Error;
+
+    use crate::utils::{get_full_path_for_resource_file, CommandTestRunner};
+
+    #[derive(Default)]
+    struct TestCommandTestRunner<'args> {
```

Comment:
Interesting! Adding these in commands makes testing so much easier! Also good to see the builder pattern in Rust!